sdk/src/main/java/com/google/cloud/dataflow/sdk/util/BigQueryTableRowIterator.java

```diff
@@ -435,11 +463,11 @@ private TableReference executeQueryAndWaitForCompletion()
         result = client.execute();
         break;
       } catch (IOException e) {
-        LOG.error(String.format(error, errorArgs), e.getMessage());
+        String errorMessage = error + String.format(" Cause: %s", e.getMessage());
+        LOG.error(errorMessage);
         if (!BackOffUtils.next(sleeper, backOff)) {
-          LOG.error(
-              String.format(error, errorArgs), "Failing after retrying " + MAX_RETRIES + " times.");
-          throw e;
+          LOG.error(error + String.format(" Failing after retrying %d times.", MAX_RETRIES));
```

Comment:
and similarly something like: `LOG.error("{}. Failing to execute job after {} attempts.", error, MAX_RETRIES+1)`